pkg/sessions/utils/envelope.go

```diff
@@ -0,0 +1,26 @@
+package utils
+
+import (
+	"github.com/vmihailenco/msgpack/v4"
+)
+
+type SessionEnvelope struct {
```

Comment:
Are we cool with this structure? If so, it can be applied to other cookie areas beyond compressed cookie sessions to easily allow detections of cookie formats & versions, etc.

Detecting legacy JSON popping in is as easy as falling back to those decoding functions is the `msgpack.Unmarshal` fails when trying to get back a SessionEnvelope from a cookie.